### ISSUE TITLE
WordPressVIPMinimum.Constants.ConstantString.NotCheckingConstantName wrong

### ISSUE THREAD

**Issue by kkmuffme** [Automattic/VIP-Coding-Standards#480]
```
$key = 'hello';
if ( defined( $key ) && constant( $key ) === false ) {
```

gives error WordPressVIPMinimum.Constants.ConstantString.NotCheckingConstantName:
Constant name, as a string, should be used along with `defined()`.

which is wrong, as it is checked with defined.

**Comment by GaryJones** [comment]
Hi @kkmuffme 

While there may indeed be an area of improvement here, your example is flawed, in that `defined()` only works with constants, yet in your example you've got `$key` as a variable.

**Comment by kkmuffme** [comment]
No, `defined()` works with strings and NOT constants. If you pass a constant to defined, you check if the value of the constant is defined as a separate constant.

See:

```
$key = 'HELLO';
define( $key, 'world' );
if ( defined( $key ) ) {
    echo 'checked as string';
}

if ( defined( HELLO ) ) {
    echo 'checked as constant';
}
```

**Comment by jrfnl** [comment]
Closing as this was fixed by #444.
